qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java

```diff
@@ -198,168 +200,141 @@ public AnswerLink(final BaseEntity source, final BaseEntity target, final Answer
 	 */
 	@JsonIgnore
 	public void setAnswer(final Answer answer) {
+
 		this.setCreated(answer.getCreated());
 		this.setExpired(answer.getExpired());
 		this.setRefused(answer.getRefused());
 		this.setInferred(answer.getInferred());
 		
-		List<String> formatStrings = null;
-
-		String className = "";
-		try {
-			className = this.getAttribute().getDataType().getClassName();
-		} catch (NullPointerException e) {
-			e.printStackTrace();
+		Attribute attr = this.getAttribute();
+		DataType dataType = attr.getDataType();
+		String className = dataType.getClassName();
+		String result = answer.getValue();
+
+		if (StringUtils.isBlank(result) || ("null".equals(result) && !className.contains("String"))) {
+			valuesNull();
+			return;
 		}
 
 		switch (className) {
-			case "life.genny.qwandaq.entity":
-				List<String> beCodeList = new CopyOnWriteArrayList<String>();
-				beCodeList.add(answer.getValue());
-				setValueBaseEntityCodeList(beCodeList);
+
+			case "life.genny.qwandaq.entity.BaseEntity":
+			case "life.genny.qwanda.entity.BaseEntity":
+				setValueString(result);
 				break;
+
 			case "java.lang.Integer":
 			case "Integer":
-				String result = answer.getValue();
-				if (!StringUtils.isBlank(result)) {
-
-					final Integer integer = Integer.parseInt(result);
-					setValueInteger(integer);
-				} else {
-					setValueInteger(0);
-
-				}
-
+				Integer integer = Integer.parseInt(result);
+				setValueInteger(integer);
 				break;
+
 			case "java.time.LocalDateTime":
 			case "LocalDateTime":
-				result = answer.getValue();
-				if (!StringUtils.isBlank(result)) {
-
-					formatStrings = Arrays.asList("yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm",
-							"yyyy-MM-dd HH:mm:ss", "yyyy-MM-dd", "yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd HH:mm:ss.SSSZ");
-					for (String formatString : formatStrings) {
-						try {
-							Date olddate = new SimpleDateFormat(formatString).parse(result);
-							final LocalDateTime dateTime = olddate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
-							setValueDateTime(dateTime);
-							break;
-						} catch (java.text.ParseException e) {
-							continue;
-						}
-
+				List<String> dateTimeFormats = Arrays.asList("yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm", 
```

Comment:
Extract the dateTime formats to a separate file.

Reply:
Done